docs/usage.rst

```diff
@@ -204,6 +204,31 @@ Now we're going to get the base URL and some image links::
      'image4_thumb.jpg',
      'image5_thumb.jpg']
 
+JMESPath Selector
+-----------------
+Whenever the selected elements possess JSON format, it is possible to use  ``.jmespath()`` method from
+:class:`~parsel.selector.Selector`.
+
+For example, given the selector is parsing the following element::
+
+    >>> text = '{"user":[{"name":"A","age":"25"},{"name":"B","age":"32"},{"name":"C","age":"19"}]}'
+    >>> jmespath_selector = Selector(text=text)
+
+It is possible to extract the age of the users as an integer by using the following line of code::
+
+    >>> jmespath_selector.jmespath("user[*].age.to_number(@)").getall()
+    [25,
+     32,
+     19]
+
+JMESPath selectors can be chained with css and xpath selectors,
+as well as use `.re()` and `re_first()` and the other methods.
+
+To find out more about JMESPath, head to `JMESPath Specification`_.
+
+.. _JMESPath Specification: https://jmespath.org/specification.html
```

Comment:
Up until this section, by reading the documentation, users will have no idea of the fact that Parsel supports JMESPath selection for JSON data.

I think we need to review the documentation from the beginning, and make sure that references to JMESPath are introduced as if JMESPath support had been part of Parsel forever.

Reply:
I think this raises another scope for the documentation. It's true JMES can be presented in the current doc in the introductory section, I will fix that. Nevertheless, it may be good to make this redesign of the docs a task by itself maybe?

Reply:
I have just had a proper read of the documentation, and I see what you mean that is may not be that simple.

I agree with your initial suggestion now: let this pull request be limited to this new section _and_ updating the initial README paragraph to mention JSON and JMESPath, and let the general documentation refactoring by a separate task to be done later. I actually would like to be the one to work on that.